### PR TITLE
Graph Implementation and Traversal Algorithms (Java)

### PR DESCRIPTION
**Pull request by tensoncai**
Added an article about graph traversals - Graph Implementation and Traversal Algorithms (Java)
Also added 3 pictures of graphs.

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by hectorkambow** [comment]
Plagiarism check done

File: articles/graph-traversals-java/index.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+<br />
+<br />
+
+## How to implement a graph
+
+### Node object
+
+First, we create a Node object. It contains a value and an array list of neighboring nodes.
+
+```
```
Comment (lalith1403) [inline articles/graph-traversals-java/index.md:40]: Use the java syntax for formatting java code snippets.

File: articles/graph-traversals-java/index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+Undirected and Unweighted  |  Directed and Weighted	 |	
+:-------------------------:|:-----------------------:|
+|![undirected](undirected.png)|![directed](directed.png)
+
+<br />
+<br />
+
+## How to implement a graph
+
+### Node object
```
Comment (lalith1403) [inline articles/graph-traversals-java/index.md:36]: It is not clear whether all the code snippets belong to the same code base regarding the code snippets. Can you give some instructions regarding running the code? Anyone trying this out irrespective of experience with java should be able to run the code. I am not asking you to talk about java installation, but more about the directory structure and the file names.

File: articles/graph-traversals-java/index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        n2.addEdge(n4);
+        n3.addEdge(n1);
+        n3.addEdge(n4);
+        n3.addEdge(n5);
+        n4.addEdge(n2);
+        n4.addEdge(n3);
+        n5.addEdge(n3);
+
+        // Uncomment any method and run
+        // bfs(n0);
```
Comment (lalith1403) [inline articles/graph-traversals-java/index.md:89]: Create a new section at the end and call it outputs. Under that section, copy the outputs from the terminal and paste it as a code formatted snippet. Use ```bash to format the output as seen on the terminal. For each of the methods discussed here.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
The explanation is spot on. Additional instructions on running the code and getting the output as desired would grasp a broader audience's attention. If you could include a repl.it link, that would be amazing as well. Don't embed the iframe; share the link if possible. Finally a section on the desired outputs.

File: articles/graph-traversals-java/index.md (hunk 1 above)
Comment (tensoncai) [reply articles/graph-traversals-java/index.md:40]: Hi, I'm not sure what you mean by that!

**Comment by lalith1403** [comment]
@tensoncai Once done, update us so that we can proceed. Also, there seem to be problems with the commits. It would be advisable to create a duplicate PR and notify us once done.  Refer to the upload instructions in the main repository before submitting.

**Comment by tensoncai** [comment]
Hi, thanks for your patience. I deleted my forked repository, then went through the whole process with the upload instructions posted.

I made a second pull request. Hope it works now.
https://github.com/section-io/engineering-education/pull/937 <https://github.com/section-io/engineering-education/pull/937>

Thank you for your patience,
Tenson Cai

**Pull request closed without merging**